### DISCUSSION TITLE
Null fields bubbling up

### DISCUSSION THREAD

**Discussion by fiatjaf** [graphql/graphql-wg#1291] [Q&A] [answered]
On https://www.youtube.com/watch?v=35kCTfYjoAg there is an explanation of how to deal with null results on non-null fields, but the spec proposal doesn't mention anything about that.

Suppose you have a schema like

```graphql
type Book {
  name: String!
  price: Int
  stock: Stock
}

type Stock {
  new: StockEntry!
  used: StockEntry!
}

type StockEntry {
  warehouse1: Int!
  warehouse2: Int!
}
```

and a query like

```graphql
query {
  books {
    name
    price
    stock {
      new {
        warehouse1
        ... on StockEntry @defer {
          warehouse2
        }
      }
      used {
        warehouse1
        ... on StockEntry @defer {
          warehouse2
        }
      }
    }
  }
}
```

What should happen if `books.0.store.used.warehouse2` wrongly returns `null`? Should the payload be

```json
{
  "path": ["books", 0, "stock", "used"],
  "data": null
}
```

Or should it keep track of the last nullable field in the path and return a reference to that instead, like

```json
{
  "path": ["books", 0, "stock"],
  "data": null
}
```

**Comment by benjie** [comment, marked as answer]
Hi, I think you're talking about incremental delivery so you should probably raise the question here:

https://github.com/robrichard/defer-stream-wg/discussions

(You may want to review the existing discussions to see if it has already been answered.)
